test/Serialization/load-invalid-arch.swift

```diff
@@ -0,0 +1,12 @@
+
+// RUN: %empty-directory(%t)
+// RUN: mkdir %t/new_module.swiftmodule
+// RUN: touch %t/new_module.swiftmodule/arm64.swiftmodule
+// RUN: touch %t/new_module.swiftmodule/powerpc64.swiftmodule
+// RUN: touch %t/new_module.swiftmodule/arm64.swiftdoc
+// RUN: touch %t/new_module.swiftmodule/powerpc64.swiftdoc
+// RUN: %target-swift-frontend %s -typecheck -I %t -verify -show-diagnostics-after-fatal
```

Comment:
Hm, I think we're also going to want this check for frameworks. Those have a layout of "_MyFramework_.framework/Modules/_MyFramework_.swiftmodule/_arch_.swiftmodule". (As you can see in the code, framework targets never use single-architecture modules.)